Framework/Geometry/src/RandomPoint.cpp

```diff
@@ -26,11 +26,20 @@ Kernel::V3D localPointInCylinder(const Kernel::V3D &basis,
                                  const Kernel::V3D &alongAxis,
                                  double polarAngle, double radialLength) {
   using boost::math::pow;
-  Mantid::Kernel::V3D basis2{1., 0., 0.};
-  if (basis.X() != 0. && basis.Z() != 0) {
-    const auto inverseXZSumSq = 1. / (pow<2>(basis.X()) + pow<2>(basis.Z()));
-    basis2.setX(std::sqrt(1. - pow<2>(basis.X()) * inverseXZSumSq));
-    basis2.setZ(basis.X() * std::sqrt(inverseXZSumSq));
+  // Use basis to get a second perpendicular vector to define basis2
+  Mantid::Kernel::V3D basis2;
+  if (basis.X() == 0) {
+    Mantid::Kernel::V3D x{1., 0., 0.};
+    basis2 = x;
```

Comment:
Why not just `basis.setX(1.);`? It avoid constructing and copying an entire `V3D`. Ditto for the other cases below.